cds_ils/importer/eitems/importer.py

```diff
@@ -0,0 +1,116 @@
+import uuid
+
+from invenio_app_ils.eitems.api import EItem, EItemIdProvider
+from invenio_app_ils.eitems.search import EItemSearch
+from invenio_app_ils.proxies import current_app_ils
+from invenio_db import db
+
+from cds_ils.importer.config import CDS_ILS_IMPORTER_PROVIDERS
+from cds_ils.importer.eitems.api import get_eitems_for_document_by_provider
+
+
+class EItemImporter(object):
+
+    def __init__(self, json_metadata, metadata_provider,
+                 current_provider_priority, delete_lower_priority_providers,
+                 open_access, login_required):
+        self.json_data = json_metadata
+        self.metadata_provider = metadata_provider
+        self.current_provider_priority = current_provider_priority
+        self.delete_lower_priority_providers = delete_lower_priority_providers
+        self.open_access = open_access
+        self.login_required = login_required
+
+    def _get_record_import_provider(self, record):
+        """Get an import provider of a given document."""
+        is_imported = record["created_by"]["type"] == 'import'
+        if is_imported:
+            return record["created_by"]["value"]
+
+    def _set_record_import_source(self, record_dict):
+        record_dict["created_by"] = {"type": "import",
+                                     "value": self.metadata_provider}
+
+    def _should_overwrite_eitem(self, eitem):
+        """Decide if the eitems ."""
+
+        existing_provider = self._get_record_import_provider(eitem)
+
+        if not existing_provider:
+            return False
+
+        existing_priority = CDS_ILS_IMPORTER_PROVIDERS[
+            existing_provider]['priority']
+
+        # existing_priority = 0, self.priority = 1, returns False
+        return existing_priority >= self.current_provider_priority
+
+    def update_eitems(self, matched_document):
+        """Update eitems for given document."""
+        import ipdb; ipdb.set_trace()
+        document_pid = matched_document["pid"]
+
+        results = get_eitems_for_document_by_provider(document_pid,
+                                                      self.metadata_provider)
+
+        if len(results) == 0:
+            self.import_eitem(matched_document)
+        elif len(results) == 1:
+            existing_eitem = EItem.get_record_by_pid(results.hits[0].pid)
+            metadata_to_update = {}
+            self._build_eitem_dict(
+                {}, matched_document["pid"])
+            existing_eitem.update(metadata_to_update)
+            existing_eitem.commit()
+        else:
+            for hit in results:
+                existing_eitem = EItem.get_record_by_pid(hit.pid)
+                existing_eitem.delete()
+            self.import_eitem(matched_document)
+
+        if self.delete_lower_priority_providers:
+            document_eitems = EItemSearch().search_by_document_pid(
```

Comment:
It is probably preferable to search in the PID Store db tables when using the PID directly, not ES.
Please double check this in the entire code.

Reply:
for the EItems we are not minting the `document_pid` as PID, I'm not sure if I understand your comment